src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/Gen.java

```diff
@@ -2414,6 +2430,7 @@ public boolean genClass(Env<AttrContext> env, JCClassDecl cdef) {
             toplevel = null;
             endPosTable = null;
             nerrs = 0;
+            qualifiedSymbolCache = null;
```

Comment:
I meant `qualifiedSymbolCache.clear()` instead of assigning null to it

Reply:
If we use `qualifiedSymbolCache.clear()`, maybe it is good to remove the following code currently. 

```
if (qualifiedSymbolCache == null) {
                qualifiedSymbolCache = new HashMap<>();
}
```

If the `qualifiedSymbolCache` is revised elsewhere in the future, more attention should be paid.

Reply:
Fixed.